File: sqlite-android-studio-plugin/src/main/java/com.alecstrong.sqlite.android/lang/SqliteFindUsagesProvider.java
package com.alecstrong.sqlite.android.lang;

import com.alecstrong.sqlite.android.psi.ColumnNameElement;
import com.intellij.lang.cacheBuilder.WordsScanner;
import com.intellij.lang.findUsages.FindUsagesProvider;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public final class SqliteFindUsagesProvider implements FindUsagesProvider {
  @Nullable @Override public WordsScanner getWordsScanner() {
    return null;
  }

  @Override public boolean canFindUsagesFor(@NotNull PsiElement psiElement) {
    return psiElement instanceof ColumnNameElement;
  }

  @Nullable @Override public String getHelpId(@NotNull PsiElement psiElement) {
    return null;
  }

  @NotNull @Override public String getType(@NotNull PsiElement element) {
    if (element instanceof ColumnNameElement) {
      return "sqlite column";
    }

    return "";
  }

  @NotNull @Override public String getDescriptiveName(@NotNull PsiElement element) {
    return element.getText();
  }

  @NotNull @Override public String getNodeText(@NotNull PsiElement element, boolean useFullName) {
    return element.getParent().getText();
  }
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/alecstrong/sqlite/android/lang/SqliteFindUsagesProvider.kt
package com.alecstrong.sqlite.android.lang

import com.alecstrong.sqlite.android.psi.SqliteElement.ColumnNameElement
import com.intellij.lang.findUsages.FindUsagesProvider
import com.intellij.psi.PsiElement

class SqliteFindUsagesProvider : FindUsagesProvider {
  override fun getWordsScanner() = null
  override fun canFindUsagesFor(psiElement: PsiElement) = psiElement is ColumnNameElement
  override fun getHelpId(psiElement: PsiElement) = null
  override fun getDescriptiveName(element: PsiElement) = element.text
  override fun getNodeText(element: PsiElement, useFullName: Boolean) = element.parent.text
  override fun getType(element: PsiElement) =
      when (element) {
        is ColumnNameElement -> "sqlite column"
        else -> ""
      }
}
